aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -168,25 +169,55 @@ Tensor& addmm_out_cuda_impl(Tensor& result, const Tensor& self, const Tensor& ma
             c10::nullopt /* pin_memory */));
   }
 
-  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(!result_->is_conj());
-
-  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, scalar_type, "addmm_cuda", [&] {
-    scalar_t alpha_val = alpha.to<scalar_t>();
-    scalar_t beta_val = beta.to<scalar_t>();
-    scalar_t* mat1_ptr = mat1_->data_ptr<scalar_t>();
-    scalar_t* mat2_ptr = mat2_->data_ptr<scalar_t>();
-    scalar_t* result_ptr = result_->data_ptr<scalar_t>();
-    at::cuda::blas::gemm<scalar_t>(
-      transpose_mat1 ? mat1_->is_conj() ? 'c' : 't' : 'n',
-      transpose_mat2 ? mat2_->is_conj() ? 'c' : 't' : 'n',
-      m, n, k,
-      alpha_val,
-      mat1_ptr, mat1_ld,
-      mat2_ptr, mat2_ld,
-      beta_val,
-      result_ptr, result_ld
+  // TODO: is it possible to combine the dispatch code?
+  if (isIntegralType(scalar_type, false)) {
```

Comment:
You could place the new code into a separate function and have in this function:
```cpp
if (is_int32_or_int64) {
call_integer_specialization();
return result;
}
```
And the diff would be cleaner and no need to make the existing function more complicated to quickly glance through.